Revit_Core_Engine/Query/DoesIntersect.cs

```diff
@@ -0,0 +1,156 @@
+                }
+            }
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Check if bounding box intersects with element.")]
+        [Input("bbox", "Bounding box to check the intersection for.")]
+        [Input("element", "Element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this BoundingBoxXYZ bbox, Element element)
+        {
+            if (bbox == null || element == null)
+            {
+                return false;
+            }
+
+            if (!bbox.Transform.IsIdentity)
+            {
+                BH.Engine.Base.Compute.RecordWarning("Intersection of the bounding boxe could not be checked. Only identity transformation is currently supported.");
+                return false;
+            }
+
+            Transform transform = element.Document.LinkTransform();
+
+            //host vs host
+            if (transform == null)
+            {
+                Outline outline = new Outline(bbox.Min, bbox.Max);
+
+                if (bbox.Transform.IsTranslation)
+                {
+                    outline.MinimumPoint += bbox.Transform.Origin;
+                    outline.MaximumPoint += bbox.Transform.Origin;
+                }
+
+                BoundingBoxIntersectsFilter bboxIntersect = new BoundingBoxIntersectsFilter(outline);
+                BoundingBoxIsInsideFilter bboxInside = new BoundingBoxIsInsideFilter(outline);
+                LogicalOrFilter bboxFilter = new LogicalOrFilter(new List<ElementFilter> { bboxIntersect, bboxInside });
+
+                return bboxFilter.PassesFilter(element);
+            }
+            //host vs link
+            else
+            {
+                Solid bboxSolid = bbox.ToSolid();
+                Solid transformedBBoxSolid = SolidUtils.CreateTransformed(bboxSolid, transform.Inverse);
+                ElementIntersectsSolidFilter intersectFilter = new ElementIntersectsSolidFilter(transformedBBoxSolid);
+                return intersectFilter.PassesFilter(element);
+            }
+        }
+
+        /***************************************************/
+        /****              Private methods              ****/
+        /***************************************************/
+
+        [Description("Check if two elements intersect.")]
+        [Input("hostElement", "First element to check the intersection for.")]
+        [Input("transElement", "Second element to check the intersection for.")]
+        [Input("transform", "Transformation of the second element.")]
+        [Output("bool", "Result of the intersect checking.")]
+        private static bool DoesIntersectWithTransform(this Element hostElement, Element transElement, Transform transform)
+        {
+
+            List<Solid> solids = transElement.Solids(new Options()).Select(x => SolidUtils.CreateTransformed(x, transform)).ToList();
```

Comment:
`CreateTransformed` could be moved inside the `foreach` loop - not sure if very relevant, but can imagine it improving the performance by a few % by not doing the transformation on all geometries, only the ones before the intersection